crates/primitives/src/header.rs

```diff
@@ -574,6 +577,67 @@ impl Decodable for Header {
     }
 }
 
+/// A list of [`Header`]s.
+#[main_codec]
+#[derive(Debug, Default, PartialEq, Eq, Clone, RlpEncodableWrapper, RlpDecodableWrapper)]
+pub struct Headers(Vec<Header>);
+
+impl Headers {
+    /// Create a new Headers instance.
+    pub fn new(headers: Vec<Header>) -> Self {
+        Headers(headers)
+    }
+
+    /// Get an iterator over the headers.
+    pub fn iter(&self) -> std::slice::Iter<'_, Header> {
+        self.0.iter()
+    }
+
+    /// Get a mutable iterator over the headers.
+    pub fn iter_mut(&mut self) -> std::slice::IterMut<'_, Header> {
+        self.0.iter_mut()
+    }
+
+    /// Convert [Headers] into a Vec<[Header]>
+    pub fn into_vec(self) -> Vec<Header> {
+        self.0
+    }
+}
+
+impl IntoIterator for Headers {
+    type Item = Header;
+    type IntoIter = std::vec::IntoIter<Header>;
+
+    fn into_iter(self) -> Self::IntoIter {
+        self.0.into_iter()
+    }
+}
+
+impl From<Vec<Header>> for Headers {
+    fn from(headers: Vec<Header>) -> Self {
+        Headers(headers)
+    }
+}
```

Comment:
please remove all this boiler plate code for interfacing between types by using `derive_more` derive macros. with the reduced LOC, we can better see the code that adds reth specific functionality, which immensely improves readability.